File: src/main/java/com/LMS_Project/service/AuthService.java
package com.LMS_Project.service;

import com.LMS_Project.dto.request.LoginRequest;
import com.LMS_Project.dto.request.RegisterRequest;
import com.LMS_Project.dto.response.AuthResponse;

public interface AuthService {

	 AuthResponse register(RegisterRequest request);
	    AuthResponse login(LoginRequest request);
}

File: lms-frontend/src/services/auth.service.ts
import axiosInstance from "./axios.config";

export interface LoginRequest {
  username: string;
  password: string;
}

export interface RegisterRequest {
  username: string;
  password: string;
}

export interface AuthResponse {
  token:    string;
  username: string;
  role:     string;
  message:  string;
}

export interface ApiResponse<T> {
  message: string;
  data:    T;
  errors?: Record<string, string>;
}

export const authService = {
  login: async (data: LoginRequest) => {
    const res = await axiosInstance.post<ApiResponse<AuthResponse>>(
      "/api/auth/login",
      data
    );
    return res.data;
  },

  register: async (data: RegisterRequest) => {
    const res = await axiosInstance.post<ApiResponse<AuthResponse>>(
      "/api/auth/register",
      data
    );
    return res.data;
  },
};